app/v1/services/AmendCustomEmploymentService.scala

```diff
@@ -33,20 +33,37 @@ class AmendCustomEmploymentService @Inject() (connector: AmendCustomEmploymentCo
   def amendEmployment(request: AmendCustomEmploymentRequest)(implicit ctx: RequestContext, ec: ExecutionContext): Future[ServiceOutcome[Unit]] =
     connector.amendEmployment(request).map(_.leftMap(mapDownstreamErrors(downstreamErrorMap)))
 
-  private val downstreamErrorMap: Map[String, MtdError] =
-    Map(
+  private val downstreamErrorMap: Map[String, MtdError] = {
+    val ifsErrors = Map(
       "INVALID_TAXABLE_ENTITY_ID" -> NinoFormatError,
-      "INVALID_TAX_YEAR"          -> TaxYearFormatError,
-      "INVALID_EMPLOYMENT_ID"     -> EmploymentIdFormatError,
-      "NOT_SUPPORTED_TAX_YEAR"    -> RuleTaxYearNotEndedError,
-      "INVALID_DATE_RANGE"        -> RuleStartDateAfterTaxYearEndError,
-      "INVALID_CESSATION_DATE"    -> RuleCessationDateBeforeTaxYearStartError,
-      "CANNOT_UPDATE"             -> RuleUpdateForbiddenError,
-      "NO_DATA_FOUND"             -> NotFoundError,
-      "INVALID_PAYLOAD"           -> InternalError,
-      "INVALID_CORRELATIONID"     -> InternalError,
-      "SERVER_ERROR"              -> InternalError,
-      "SERVICE_UNAVAILABLE"       -> InternalError
+      "INVALID_TAX_YEAR" -> TaxYearFormatError,
+      "INVALID_EMPLOYMENT_ID" -> EmploymentIdFormatError,
+      "NOT_SUPPORTED_TAX_YEAR" -> RuleTaxYearNotEndedError,
+      "INVALID_DATE_RANGE" -> RuleStartDateAfterTaxYearEndError,
+      "INVALID_CESSATION_DATE" -> RuleCessationDateBeforeTaxYearStartError,
+      "CANNOT_UPDATE" -> RuleUpdateForbiddenError,
+      "OUTSIDE_AMENDMENT_WINDOW" -> RuleOutsideAmendmentWindowError,
+      "NO_DATA_FOUND" -> NotFoundError,
+      "INVALID_PAYLOAD" -> InternalError,
+      "INVALID_CORRELATIONID" -> InternalError,
+      "SERVER_ERROR" -> InternalError,
+      "SERVICE_UNAVAILABLE" -> InternalError
     )
 
+    val hipErrors: Map[String, MtdError] = Map(
```

Comment:
Error code 5000 -> RULE_TAX_YEAR_NOT_SUPPORTED is missing

Reply:
Even though the 501 will get captured and returned as a 500 so we'll never get that far?

Reply:
We have added the 501 error [here](https://github.com/hmrc/individuals-employments-income-api/blob/main/app/shared/connectors/httpparsers/StandardDownstreamHttpParser.scala#L65) so it will get mapped instead of defaulting to 500

Reply:
So if the code `5000` is mapped to a 400 `RULE_TAX_YEAR_NOT_SUPPORTED` in the service and there is a case where it is returned, it will return `RULE_TAX_YEAR_NOT_SUPPORTED`. We have added [NOT_IMPLEMENTED](https://github.com/hmrc/individuals-employments-income-api/blob/main/app/shared/connectors/httpparsers/StandardDownstreamHttpParser.scala#L65) to make sure the mapping is possible. Also, a test can be done in the stub to simulate the behaviour using a GTS. In terms of the 401, we didn't add it as it will always be mapped to 500 so there is no point adding an error that will always be 500 in the SHP.

Reply:
It is already added on main so you just need a rebase.